wp/lib/bap_wp/src/z3_utils.ml

```diff
@@ -111,44 +111,40 @@ let mk_smtlib2_single (env : Env.t) (smt_post : string) : Constr.t =
   let ctx = Env.get_context env in
   mk_smtlib2 ctx smt_post decl_syms
 
-let construct_pointer_constraint (l: string list) (env1 : Env.t)
-    (env2: Env.t option) : Constr.t =
+let construct_pointer_constraint (l_orig : Constr.z3_expr list) (env1 : Env.t)
+    (l_mod : (Constr.z3_expr list) option) (env2: Env.t option) : Constr.t =
   let stack_end = Env.get_stack_end env1 in
   let ctx = Env.get_context env1 in
   let arch = match Env.get_arch env1 |> Bap.Std.Arch.addr_size with
     | `r32 -> 32
     | `r64 -> 64 in
   let rsp = Env.get_sp_name env1 in
-  let function_name = "construct_pointer_constraint" in
-  let err_msg_input = "invalid register name in pointer register list" in
+  let function_name = "construct_pointer_constraint: " in
   let err_msg_rsp = "stack pointer not found" in
   let sb_bv = Z3.BitVector.mk_numeral ctx (stack_end |> Int.to_string) arch in
-  let gen_constr = match env2 with
+  let gen_constr, l = match l_mod, env2 with
     (* comparative case *)
-    | Some env2 ->
+    | Some l_mod, Some env2 ->
       let init_var_map_orig = Env.get_init_var_map env1 in
       let init_var_map_mod = Env.get_init_var_map env2 in
-      (* we do want exceptions here if the register names are invalid
-       *  or RSP doesn't exist *)
-      let rsp_orig = Option.value_exn ~message:(function_name ^ ": original " ^ err_msg_rsp)
+      (* we do want exceptions here if RSP doesn't exist *)
+      let rsp_orig = Option.value_exn
```

Comment:
Does this error here if I have a function that doesn't use RSP? For example, something like

```
sub():
RDI := 0xDEADBEEF
call RDI with noreturn
```

Reply:
Yes, I just tried and this does error. Perhaps the right thing to do here is to not error, generate an `true` constraint, and print a warning to the log.

Reply:
What do you do for other registers that can't be found in the subroutine?

Reply:
For the other registers, if they do not exist, I create them, create their respective constraint (which ends up being trivial since they are unused in the sub), and print a warning to the log. To be consistent, I can also do this with the stack pointer.

Reply:
Since RSP is used in the constraint generated, if we do create this trivial constraint, is it equivalent to true? I'm actually having a hard time trying to figure out which is the final constraint generated from the comments. Can you enlighten me haha?

Reply:
Suppose, for some register `R`, we generate this constraint on a single binary (the comparative case is analogous). Then we would have the constraint be:

```
R >= RSP \/ R <= stack_bottom
```
Since `RSP` can be chosen to be anything (because it's not used in the sub), `CBAT` can choose it to be `stack_bottom`. I think this evaluates to `true` in that case. The final constraint is a bunch of these constraints (`R` is the only thing that changes) anded together, which should remain as true.

Reply:
Okay! And `RSP <= RSP` is true, so I'm convinced this is true. Go ahead and place it with the other variables to be more consistent and add a big warning log for this.